stdlib/Random/docs/src/index.md

```diff
@@ -26,6 +26,9 @@ unbounded integers, the interval must be specified (e.g. `rand(big.(1:6))`).
 Additionally, normal and exponential distributions are implemented for some `AbstractFloat` and
 `Complex` types, see [`randn`](@ref) and [`randexp`](@ref) for details.
 
+!!! warn
+    Because the precise way in which random numbers are generated is considered an implementation detail, bug fixes and speed improvements may change the stream of numbers that are generated after a version change. Relying on a specific seed or generated stream of numbers during unit testing is thus discouraged - consider testing properties of the methods in question instead.
```

Comment:
Maybe specify "minor version change" ? (someone else should confirm)

Reply:
I guess that depends on if RNG is only allowed to change in a minor but not a bug version?

Reply:
Ah good point; improvements changes will happen in minor versions (and be labelled as "minor change"), but a bug fix changing the streams (although rare) might happen in a patch version.